src/pages/view/FilterBar/SaveFilterDialog.js

```diff
@@ -0,0 +1,89 @@
+import {
+    Button,
+    Modal,
+    ModalContent,
+    ModalActions,
+    ButtonStrip,
+    ModalTitle,
+    InputField,
+    SingleSelectOption,
+    SingleSelectField,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import i18n from '@dhis2/d2-i18n'
+import {
+    filterVisibility, privateVisiblity,
+} from '../../../reducers/savedFilters'
+
+const SaveFilterDialog = ({
+    onCancel,
+    onConfirm,
+    filter = {},
+    showScope = true,
+}) => {
+    const [visibility, setVisibility] = React.useState(filter.visibility || privateVisiblity)
+    const [name, setName] = React.useState(filter.id ? filter.name : '')
+
+    const handleVisibilityChange = ({ selected }) => {
+        setVisibility(selected)
+    }
+    const handleNameChange = ({ value }) => {
+        setName(value)
+    }
+    const handleConfirm = () => {
+        onConfirm({ name, visibility, id: filter.id })
+    }
+
+    return (
+        <Modal
+            dataTest="save-filter-dialog"
+            onClose={onCancel}
+            small
+            position="middle"
+        >
+            <ModalTitle>{i18n.t('Save Filter')}</ModalTitle>
+            <ModalContent>
+                <InputField
+                    type="text"
+                    placeholder={i18n.t('Untitled filter')}
+                    label={i18n.t('Name')}
+                    value={name}
+                    onChange={handleNameChange}
+                />
+                {showScope && (
+                    <SingleSelectField
+                        label={i18n.t('Scope')}
+                        selected={visibility}
+                        onChange={handleVisibilityChange}
+                    >
+                        {filterVisibility.map((f) => (
```

Comment:
Not a big deal, but let's try to avoid one letter variables. `f` -> `filter`, maybe?